libs/common-traits/src/lib.rs

```diff
@@ -278,3 +278,183 @@ impl<T> PreConditions<T> for Never {
 pub trait TrancheToken<PoolId, TrancheId, CurrencyId> {
 	fn tranche_token(pool: PoolId, tranche: TrancheId) -> CurrencyId;
 }
+
+pub mod fees {
+	use frame_support::dispatch::DispatchResult;
+	use frame_support::traits::tokens::Balance;
+
+	/// A way to identify a fee value.
+	pub enum Fee<Balance, FeeKey> {
+		/// The fee value itself.
+		Balance(Balance),
+
+		/// The fee value is already stored and identified by a key.
+		Key(FeeKey),
+	}
+
+	/// A trait that used to deal with fees
+	pub trait Fees<K> {
+		type AccountId;
+		type Balance: Balance;
+
+		/// Get the fee balance for a fee key
+		fn fee_value(key: K) -> Self::Balance;
+
+		/// Pay an amount of fee to the block author
+		/// If the `from` account has not enough balance or the author is invalid the fees are not
+		/// paid.
+		fn fee_to_author(from: &Self::AccountId, fee: Fee<Self::Balance, K>) -> DispatchResult;
+
+		/// Burn an amount of fee
+		/// If the `from` account has not enough balance the fees are not paid.
+		fn fee_to_burn(from: &Self::AccountId, fee: Fee<Self::Balance, K>) -> DispatchResult;
+
+		/// Send an amount of fee to the treasury
+		/// If the `from` account has not enough balance the fees are not paid.
+		fn fee_to_treasury(from: &Self::AccountId, fee: Fee<Self::Balance, K>) -> DispatchResult;
+	}
+
+	pub struct NoFees<AccountId, Balance>(sp_std::marker::PhantomData<(AccountId, Balance)>);
+
+	impl<K, A, B: Balance> Fees<K> for NoFees<A, B> {
+		type AccountId = A;
+		type Balance = B;
+
+		fn fee_value(_: K) -> Self::Balance {
+			Self::Balance::default()
+		}
+
+		fn fee_to_author(_: &Self::AccountId, _: Fee<Self::Balance, K>) -> DispatchResult {
+			Ok(())
+		}
+
+		fn fee_to_burn(_: &Self::AccountId, _: Fee<Self::Balance, K>) -> DispatchResult {
+			Ok(())
+		}
+
+		fn fee_to_treasury(_: &Self::AccountId, _: Fee<Self::Balance, K>) -> DispatchResult {
+			Ok(())
+		}
+	}
+
+	#[cfg(feature = "std")]
+	pub mod test_util {
```

Comment:
This is a utility to easily mock the Fees from the pallets. It has not production code.